### ISSUE TITLE
'memory' must be strictly typed

### ISSUE THREAD

**Issue by yegor256** [objectionary/eo#761]
Let's make sure `memory` can only accept one "type" of data primitives. In other words, this code should lead to an error:

```
memory 42 > m
m.write 3.1416
```

At the first line we save `int` and at the second one attempt to save `float`. The second line should cause an error.

Let's not forget to update the "Origins of Objects" paper.

p.s. Moreover, let's implement `memory` in EO instead of Java and let it use `ram`. This will mean that it will fail with "not enough memory" if we make an attempt to store a longer string into it:

```
memory "Hello" > m
m.write "Much longer string!"
```

**Comment by yegor256** [comment]
@Graur can you please help

**Comment by Graur** [comment]
@yegor256 How can we check types in EO? By size?

**Comment by yegor256** [comment]
@Graur yes, just checking by size will be enough

**Comment by Graur** [comment]
@MCJOHN974 Can you help us here?

**Comment by Graur** [comment]
@MCJOHN974 What about the progress of this task? Do you need some help?

**Comment by yegor256** [comment]
@Graur what's up with this?

**Comment by Graur** [comment]
@yegor256 How do you think, do we need this changes right now? They will break a lot of our existing code. I asked because I wrote some tests and understood the consequences

**Comment by yegor256** [comment]
@Graur really "a lot"? let's disable the tests that fail and add puzzles. This change is very important, we must introduce it.

**Comment by 0pdd** [comment]
@yegor256 the puzzle [#2211](https://github.com/objectionary/eo/issues/2211) is still not solved.

**Comment by 0pdd** [comment]
@yegor256 3 puzzles [#2211](https://github.com/objectionary/eo/issues/2211), [#2434](https://github.com/objectionary/eo/issues/2434), [#2435](https://github.com/objectionary/eo/issues/2435) are still not solved.

**Comment by 0pdd** [comment]
@yegor256 2 puzzles [#2211](https://github.com/objectionary/eo/issues/2211), [#2434](https://github.com/objectionary/eo/issues/2434) are still not solved; solved: [#2435](https://github.com/objectionary/eo/issues/2435), [#2452](https://github.com/objectionary/eo/issues/2452).
